packages/kit/src/core/adapt/prerender.js

```diff
@@ -143,7 +143,7 @@ export async function prerender({ cwd, out, log, config, build_data, fallback, a
 			const is_html = response_type === REDIRECT || type === 'text/html';
 
 			const parts = path.split('/');
-			if (is_html && parts[parts.length - 1] !== 'index.html') {
+			if (is_html && !parts[parts.length - 1].endsWith('.html')) {
```

Comment:
why did this need to change? if you encounter `about.html` why are you pushing `index.html`?

Reply:
Before this change, paths like `**/about.html` are modified to `**/about.html/index.html` so I changed not to do so.